### PR TITLE
Immutable versions

### PR DESCRIPTION
**Pull request by umitkavala**
Updated to immutable records: Cluster\Memberlist Cluster\Member\SimpleMemberStrategy

## Description

https://github.com/AsynkronIT/protoactor-dotnet/issues/561 

## Purpose
This pull request is a:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/Proto.Cluster/Member/MemberList.cs [hunk 1]
```diff
@@ -228,13 +231,13 @@ private void MemberLeave(Member memberThatLeft)
...
-                    _memberStrategyByKind.Remove(k);
+                    MemberStrategyByKind.Remove(k);
                 }
             }
 
-            _bannedMembers.Add(memberThatLeft.Id);
+            BannedMembers.Add(memberThatLeft.Id);
             
-            _members.Remove(memberThatLeft.Id);
+            Members.Remove(memberThatLeft.Id);
```
Comment (rogeralsing) [inline src/Proto.Cluster/Member/MemberList.cs]: As this is an immutable object, Remove only removes the value from the immutable copy.

It needs to be this for every place where members are added or removed
```csharp
Members = Members.Remove(...)
```

Comment (umitkavala) [reply src/Proto.Cluster/Member/MemberList.cs]: Agree. Changes applied.

**Comment by rogeralsing** [comment]
It looks good now!
There are some small merge conflicts still there

**Comment by rogeralsing** [comment]
Many thanks!

**Comment by umitkavala** [comment]
> Many thanks!

build is broken. There is some issue on sln file. I'm looking right now.
Is everything ok?

**Comment by umitkavala** [comment]
LocalAffinityStrategy is new class and implement IMemberStrategy. Its broken because of implementation needs to change.

**Comment by umitkavala** [comment]
I've fixed LocalAffinityStrategy problem. And also did mistake when merge on MemberList. you had new Cluster parameter which I didn't recognize. Its on new PR.
sorry for inconvenience.

**Pull request merged**